content/en/docs/tasks/debug-application-cluster/inside-container-developing.md

```diff
@@ -0,0 +1,91 @@
+---
+title: Develop and debug the services in the container
+content_type: task
+---
+
+Kubernetes applications are generally composed of many microservices. Due to environment differences and resource constraints, these microservices somtimes can not run locally as source code. Therefore, developers need to rebuild the image, modify the image version of the workload and then wait for the Pod scheduling after coding. This causes the low efficiency of development in Kubernetes emvironment.
+
+Nocalhost is a tool that enables developers to develop applications in Kubernetes Cluster. It can transfer the remote workload to development mode by using Nocalhost IDE (including VSCode and Jetbrains). When the developer writes code locally, any code changes will be automatically synchronized to the remote development container and initiate the application update immediately (depending on the hot load mechanism of the application or restart the application). The development container inherits all the declarative configurations (configmap, secret, volume, Env) from the original workload, which provides the basis for the application to run in the development container as source code.
+
+This is an tutorial of developing services in the container in Kubernetes environment using  `Nocalhost` .
+
+
+
+## {{% heading "prerequisites" %}}
+
+* Kubernetes cluster is installed
+*  [Nocalhost](https://nocalhost.dev/installation/) extension installed (supporting VSCode and Jetbrains)
+
+## Add Kubeconfig for Nocalhost 
+
+We take the VSCode extension as an example here. Open the Nocalhost extension, then you can add the Kubernetes Cluster by selecting the context in `~/.kube/config` or copying the "kubeconfig".  
```

Comment:
I wouldn't be confident I could explain these steps to someone else. What should a reader do?